### ISSUE TITLE
Random dropped frames but `stats.lua` doesn't pick it up

### ISSUE THREAD

**Issue by Obegg** [mpv-player/mpv#14245]
### mpv Information

```bash
mpv v0.38.0-353-g021c5ded Copyright © 2000-2024 mpv/MPlayer/mplayer2 projects
 built on May 26 2024 15:03:59
libplacebo version: v7.349.0 (v6.338.0-133-g9e1257c-dirty)
FFmpeg version: N-115391-g63697d335
FFmpeg library versions:
   libavutil       59.20.100
   libavcodec      61.5.104
   libavformat     61.3.104
   libswscale      8.2.100
   libavfilter     10.2.102
   libswresample   5.2.100
```

### Other Information

- Windows version: Microsoft Windows [Version 10.0.19044.4412]
- GPU model, driver and version: NVIDIA GeForce GTX 1650, 555.85 (Using [NVCleanstall](https://www.techpowerup.com/download/techpowerup-nvcleanstall/))
- Source of mpv: https://github.com/zhongfly/mpv-winbuild/releases
- Introduced in version: Don't know

### Reproduction Steps

```
mpv --no-config t.m2ts --fs --video-sync=display-resample --log-file=t.txt --vo=gpu-next --profile=high-quality --hwdec=yes --gpu-debug
```

### Expected Behavior

No random visual frame drops, but if a frame were to drop - correctly pick it up on `stats.lua` and on log file.

### Actual Behavior

Random visual frame drops, but `stats.lua` doesn't pick it up (dropped frames show 0, same with mistimed and delayed) and log file doesn't pick it up.
I checked the logs and no error appears other than `[vo/gpu-next/libplacebo] (Re)creating 2048x1768x0 texture with format r8: unknown`.

### Log File

[t.txt](https://github.com/mpv-player/mpv/files/15454786/t.txt) - Using OSD to monitor dropped frames
[t2.txt](https://github.com/mpv-player/mpv/files/15454800/t2.txt) - Not using OSD at all (still drops frames)

### I carefully read all instruction and confirm that I did the following:

- [X] I tested with the latest mpv version to validate that the issue is not already fixed.
- [X] I provided all required information including system and mpv version.
- [X] I produced the log file with the exact same set of files, parameters, and conditions used in "Reproduction Steps", with the addition of `--log-file=output.txt`.
- [X] I produced the log file while the behaviors described in "Actual Behavior" were actively observed.
- [X] I attached the full, untruncated log file.
- [X] I attached the backtrace in the case of a crash.

**Comment by llyyr** [comment]
> I _think_ the problem was because I installed mpv on `C:\Program Files\mpv`, now it's on desktop and having no issues so far, will re-open this issue if I have the same issue.

Why would the install location matter?

**Comment by Jules-A** [comment]
Umm... You didn't actually mention if software decoding fixed it or not?

For the last month or so I've been experiencing a bug where if I alt-tab away while anything is using a decent amount of IO, MPV will freeze for 15-30secs and claim that shader compilation took that long, when it returns it will say around 150 dropped frames but it should be way higher.
Have you monitored your disk usage while MPV is playing? Have you tried setting a larger cache?

**Comment by llyyr** [comment]
> Before I assume it's hardware or firmware - I suspect it's the files, could anyone kindly provide me with a good video file to test stuttering on 23.976fps content? I already have a test file but I would like something else

https://github.com/mpv-player/random-stuff/tree/master/vsync

**Comment by kasper93** [comment]
Does `gpu-api=vulkan` works better?

**Comment by kasper93** [comment]
Did it start recently? Could you try some older builds to see where the issue started?

There is something wrong with a/v sync code when the estimated vsync value is used and is changing. The bug is mostly hidden, if the estimated value is not used. It would be good if you were able to pinpoint regression range.

Could you check also this build https://github.com/mpv-player/mpv/pull/14286 ? This has nasty way of accumulating error unfortunately, even though one would think otherwise, the values are not that useful. But this shouldn't cause a problem, the underlying algorithm should be resilient to any vsync values and resync if needed. Else we are just hiding bugs.

Either way, I have more changes in this area that I need to finish, but this is tricky to debug and validate.

**Comment by the-database** [comment]
I have been running into the same exact issue recently, just in the past few weeks or so, having changed nothing about mpv in several months. 

- Windows version: Microsoft Windows [Microsoft Windows 11 Pro  10.0.22631]
- GPU model, driver and version: NVIDIA GeForce GTX 1650, 555.85 
- Source of mpv: https://github.com/zhongfly/mpv-winbuild/releases

I just tried using DDU to uninstall my NVIDIA driver, and then installed an older NVIDIA driver version, 551.86. And now I am not seeing any issues at all so far, but as stated above it's hard to tell if the issue is truly fixed until I've used it for a while. After using it more I'll report back whether the issue seems resolved or if it happens again.

**Comment by kasper93** [comment]
There are a lot of variables at play here. The issue itself is not easy to reproduce, not always obvious to notice. We have to be systematic about, else we will not get anywhere.

What we need
1. Narrow down mpv version when the regression started. Or even check if it is regression in the first place.
2. Gather information about the issue. 
    - @Obegg says it is not reproducible with Vulkan
    - @the-database says Vulkan is also affected
    - This makes me think we are not looking at the exactly the same issue. Although it is hard to say, because it may be generic A/V issue that is exposed by different presentation timings.

Currently I'm not sure where to look for. I myself don't see the issue. Only dropped frames I get is when it indeed fails to render in time due to some other factors, but in normal playback I have not notice a problem. We need your help to narrow down the issue.

**Comment by the-database** [comment]
I would add that in one of my tests where it happened with vulkan, I was watching a 20 minute video and had around 15 minutes of perfectly smooth playback. Then it started happening at around the 15 minute mark of the video, and then once it started happening it would happen consistently on that video and other vidoes as well.

**Comment by kasper93** [comment]
>  Is this error ever going to get fixed? I remember seeing it a while ago when I was testing some PRs, at the time it was spamming the console, this time it doesn't spam the console but it still exists.

MSVC build has lcms2 included, currently it is not available, because of one external repository provider is upgrading their gitlab instance, should be back tomorrow.

**Comment by the-database** [comment]
> Wait, are we talking about the same issue? I'm talking about dropped frames which are not reported to `stats.lua`, effectivily rendering `stats.lua` useless and can only rely on your eyes to tell if frames have been dropped or not (which is quite hard, need a test file and need to stare at the screen for long durations)

I think we're talking about two different issues. Your issue was unreported dropped frames while using d3d11, which looks like it has been fixed in mpv. My issue is unreported dropped frames while using vulkan. I've done some more testing and here is what I found: 

- While running NVIDIA driver version 551.86 for the past few days, I have not encountered unreported dropped frames while using vulkan at all. 
- Today I tried upgrading to driver version 555.85. I encountered unreported dropped frames after that. 
- Then I upgraded the newly released driver version 555.99. I still encountered unreported dropped frames immediately. 
- I downgraded to driver version 551.86 (without using DDU this time). The unreported dropped frames issue was resolved immediately.  
- I noticed some reports of newer NVIDIA driver versions having issues related to vulkan: https://forums.libretro.com/t/nvidia-555-85-issue-with-vulkan-on-gsync/45204/4

Based on the above, I think unreported dropped frames on vulkan is not an mpv bug, but an NVIDIA driver bug, so I believe this issue can be closed.
